Global.h

```diff
@@ -543,10 +543,16 @@ namespace vstd
 		});
 	}
 
+	template<typename T>
+	T advance_r(const T &obj, int change)

```

Comment:
what does _r mean? not possible to overload advance?

Reply:
version that returns resulting value instead of modifying parameter.

Overriding should be possible but would be confusing due to having two functions with similar signature but one of which modifies passed value and another does not.

Personally I'd rather remove existing advance() and always return resulting value.
Only 8 hits in code for this function. Might be better to replace.

Reply:
- removed vstd::advance - all usages can be replaced with operator ++
- added vstd::next, based on std::next (instead of advance_r)
(will push changes later)